SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2Producer.cc

```diff
@@ -0,0 +1,185 @@
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CondFormats/EcalObjects/interface/EcalLiteDTUPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalLiteDTUPedestalsRcd.h"
+
+//#include "CondFormats/EcalObjects/interface/EcalTPGTowerStatus.h"
+//#include "CondFormats/DataRecord/interface/EcalTPGStripStatusRcd.h"
+//#include "CondFormats/EcalObjects/interface/EcalTPGStripStatus.h"
+
+#include "EcalEBTrigPrimPhase2Producer.h"
+
+#include <memory>
+
+#include "SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBPhase2TrigPrimAlgo.h"
+
+EcalEBTrigPrimPhase2Producer::EcalEBTrigPrimPhase2Producer(const edm::ParameterSet& iConfig)
+    : barrelOnly_(iConfig.getParameter<bool>("BarrelOnly")),
+      debug_(iConfig.getParameter<bool>("Debug")),
+      famos_(iConfig.getParameter<bool>("Famos")),
+      binOfMaximum_(iConfig.getParameter<int>("binOfMaximum")) {
+  tokenEBdigi_ = consumes<EBDigiCollectionPh2>(iConfig.getParameter<edm::InputTag>("barrelEcalDigis"));
+
+  eTTmapToken_ = esConsumes<edm::Transition::BeginRun>();
+  theGeometryToken_ = esConsumes<edm::Transition::BeginRun>();
+
+  theEcalTPGPedestals_Token_ =
+      esConsumes<EcalLiteDTUPedestalsMap, EcalLiteDTUPedestalsRcd, edm::Transition::BeginRun>();
+  theEcalEBPhase2TPGPedestals_Token_ =
+      esConsumes<EcalEBPhase2TPGPedestalsMap, EcalEBPhase2TPGPedestalsRcd, edm::Transition::BeginRun>();
+
+  theEcalTPGCrystalStatus_Token_ =
+      esConsumes<EcalTPGCrystalStatus, EcalTPGCrystalStatusRcd, edm::Transition::BeginRun>();
+  theEcalEBPhase2TPGLinearization_Token_ =
+      esConsumes<EcalEBPhase2TPGLinearizationConst, EcalEBPhase2TPGLinearizationConstRcd, edm::Transition::BeginRun>();
+  theEcalEBTPGAmplWeightIdMap_Token_ =
+      esConsumes<EcalEBPhase2TPGAmplWeightIdMap, EcalEBPhase2TPGAmplWeightIdMapRcd, edm::Transition::BeginRun>();
+  theEcalEBTPGTimeWeightIdMap_Token_ =
+      esConsumes<EcalEBPhase2TPGTimeWeightIdMap, EcalEBPhase2TPGTimeWeightIdMapRcd, edm::Transition::BeginRun>();
+  theEcalTPGWeightGroup_Token_ = esConsumes<EcalTPGWeightGroup, EcalTPGWeightGroupRcd, edm::Transition::BeginRun>();
+
+  //register your products
+  produces<EcalEBPhase2TrigPrimDigiCollection>();
+}
+
+void EcalEBTrigPrimPhase2Producer::beginRun(edm::Run const& run, edm::EventSetup const& setup) {
+  auto const& theGeometry = setup.getData(theGeometryToken_);
+  auto const& eTTmap = setup.getData(eTTmapToken_);
+
+  algo_ = std::make_unique<EcalEBPhase2TrigPrimAlgo>(&eTTmap, &theGeometry, binOfMaximum_, debug_);
+
+  // get a first version of the records
+  cacheID_ = this->getRecords(setup);
+
+  nEvent_ = 0;
+}
+
+unsigned long long EcalEBTrigPrimPhase2Producer::getRecords(edm::EventSetup const& setup) {
+  // get parameter records for xtals
+  auto theEcalEBPhase2TPGLinearization_handle = setup.getHandle(theEcalEBPhase2TPGLinearization_Token_);
+  const EcalEBPhase2TPGLinearizationConst* ecaltpLin = theEcalEBPhase2TPGLinearization_handle.product();
```

Comment:
```suggestion
  const auto* ecaltpLin = &setup.getData(theEcalEBPhase2TPGLinearization_Token_);
```
There seems to be no need for using handles here and also below.

Reply:
Any advantage with one or the other ?

Reply:
If the code always gets the product from the handle (i.e. it does not deal with the case of the product being missing), avoiding the handle makes that intent more clear, and the code simpler to read.

Reply:
Please consider implementing these suggestions.